packages/commonjs-shim/src/constants.ts

```diff
@@ -0,0 +1,14 @@
+export const CJSyntaxRe = /__filename|__dirname|require\(|require\.resolve\(/;
+
+export const CJSShim = `
+// -- CommonJS Shims --
+import cjsUrl from 'url';
+import cjsPath from 'path';
+import cjsMod from 'module';
+const __filename = cjsUrl.fileURLToPath(import.meta.url);
```

Comment:
While it may seem obvious, we can't account for all the different ways this may be used and what whacky edge cases users may come up with. We should document that `__filename` is intended to contain the filename for the bundle in which the shim is inserted, and that there may be edge cases depending on its use.

Reply:
good point. I added a short description for every variable in the README.md